### PR TITLE
CASMHMS-6299: Fix PCS/TRS resource leaks and scaling issues

### PR DESCRIPTION
**Pull request by jwlv**
# NOTE TO REVIEWERS:  Final changes to this PR require TRS PR merged, then official TRS release module imported here - it will be a 3.0.0 version

## Summary and Scope

A variety of changes related to PCS scaling issues being experienced in the field.  Some details are as follows:

- Updated hms-trs-app-api vendor code (bug fixes and enhancements)
- Passed PCS's log level through to TRS to match PCS's
- Configured TRS to use connection pools for status requests to BMCs
- Renamed PCS_STATUS_HTTP_TIMEOUT to PCS_STATUS_TIMEOUT as it is not an http timeout
- Added PCS_MAX_IDLE_CONNS and PCS_MAX_IDLE_CONNS_PER_HOST env variables which allow overriding PCS's connection pool settings in TRS
- Added PCS_BASE_TRS_TASK_TIMEOUT env variable which allows the timeout for power transitions and capping to be configured
- The above variables are configurable on PCS's helm chart
- At PCS start time, log all env variables that were set
- Added PodName global to facilitate easier debug of log messages
- Log start and end of large batched requests to BMCs
- Fixed many resource leaks associated with making http requests and using TRS
- Update required version of Go to 1.23 to avoid [ https://github.com/golang/go/issues/59017](https://github.com/golang/go/issues/59017)

One thing to note about adding support for connections pools, this only provides the capability on the client side, for users of TRS.  Remote servers (or intermediate firewalls, proxies, etc) may not be configured, or desire, to maintain open idle connections for a period of time.  We discovered that while some BMCs kept connections open, others didn't.

The TRS PR these changes depends on:  [https://github.com/Cray-HPE/hms-trs-app-api/pull/11](https://github.com/Cray-HPE/hms-trs-app-api/pull/11)

The PCS helm chart PR:  [https://github.com/Cray-HPE/hms-power-control-charts/pull/38](https://github.com/Cray-HPE/hms-power-control-charts/pull/38)

## Issues and Related PRs

* Resolves [CASMHMS-6299](https://jira-pro.it.hpe.com:8443/browse/CASMHMS-6299)

## Testing

Test description:

Testing was two-pronged, unit testing (in TRS) and testing on a live system.  The majority of testing was done through the TRS unit test framework.  Extensive unit tests were added for all aspects of the connection lifecycle.

These changes were also tested alongside TRS changes on both `mug` and `rocket`.  Rocket is a CSM 1.5 system and mug is a CSM 1.6 system.  Additionally, mug was running several thousand simulated BMCs on compute nodes so that the scaling aspects of these changes could be tested.  All standard PCS operations were confirmed to continue working.

While we do not have any systems internally that are able to reproduce all of the problems at customer sites, we believe this change will go a long ways towards improving most of the issues being experienced.

Testing Check List:

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? N (not deployable on its own)
- Were continuous integration tests run? If not, why? Y
- Was upgrade tested? If not, why? N (not deployable on its own)
- Was downgrade tested? If not, why? N (not deployable on its own)
## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable

### CODE REVIEW

File: internal/domain/power-cap.go [hunk 1]
```diff
@@ -659,13 +660,23 @@ func doPowerCapTask(taskID uuid.UUID) {
...
+	// a new client or sharing (and enlarging) the PCS TRS client.
+
 	// Create TRS task list
 	trsTaskMap := make(map[uuid.UUID]model.PowerCapOperation)
 	trsTaskList := (*GLOB.RFTloc).CreateTaskList(GLOB.BaseTRSTask, len(goodOps))
 	trsTaskIdx := 0
 	for _, op := range goodOps {
 		trsTaskMap[trsTaskList[trsTaskIdx].GetID()] = op
-		trsTaskList[trsTaskIdx].RetryPolicy.Retries = 3
+		trsTaskList[trsTaskIdx].CPolicy.Retry.Retries = 3
```
Comment (mbuchmann-hpe) [inline internal/domain/power-cap.go:679]: Should this be set to the retries variable sent in?

File: internal/domain/transitions.go [hunk 2]
```diff
@@ -762,7 +772,7 @@ func doTransition(transitionID uuid.UUID) {
 			comp.Task.State = model.TaskState_Sending
 			comp.Task.Operation = powerActionOp
 			trsTaskMap[trsTaskList[trsTaskIdx].GetID()] = comp
-			trsTaskList[trsTaskIdx].RetryPolicy.Retries = 3
+			trsTaskList[trsTaskIdx].CPolicy.Retry.Retries = 3
```
Comment (mbuchmann-hpe) [inline internal/domain/transitions.go:775]: Should this be set to the retries variable sent in?

**Review by mbuchmann-hpe — APPROVED** [review]

File: internal/domain/power-cap.go (hunk 1 above)
Comment (jwlv) [reply internal/domain/power-cap.go:682]: The global variable PCS_STATUS_HTTP_RETRIES is for status requests.  At some point we may also want one for power caps and transitions.

File: internal/domain/transitions.go (hunk 2 above)
Comment (jwlv) [reply internal/domain/transitions.go:778]: The global variable PCS_STATUS_HTTP_RETRIES is for status requests.  At some point we may also want one for power caps and transitions.

File: Dockerfile
[the remark this answers is not in the document]
Comment (jwlv) [reply Dockerfile:1]: Fixed.

File: Dockerfile.integration.Dockerfile
[the remark this answers is not in the document]
Comment (jwlv) [reply Dockerfile.integration.Dockerfile:1]: Fixed.

File: Dockerfile.test.unit.Dockerfile
[the remark this answers is not in the document]
Comment (jwlv) [reply Dockerfile.test.unit.Dockerfile:1]: Fixed.

**Pull request merged**